sentry-core/src/main/java/io/sentry/core/SentryCommonOptions.java

```diff
@@ -1,263 +1,450 @@
 package io.sentry.core;
 
+import static io.sentry.core.SentryOptions.DEFAULT_DIAGNOSTIC_LEVEL;
+
 import com.jakewharton.nopen.annotation.Open;
-import java.util.ArrayList;
+import io.sentry.core.util.ReflectionUtils;
 import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
 
 /** Subset of {@link SentryOptions} properties used in 3rd party framework integrations. */
 @Open
 public class SentryCommonOptions {
 
-  /** @see SentryOptions#dsn */
-  private String dsn = "";
-
-  /** @see SentryOptions#shutdownTimeoutMillis */
-  private Long shutdownTimeoutMillis;
+  /**
+   * The DSN tells the SDK where to send the events to. If this value is not provided, the SDK will
+   * just not send any events.
+   */
+  private @Nullable String dsn;
 
-  /** @see SentryOptions#flushTimeoutMillis */
-  private Long flushTimeoutMillis;
+  /**
+   * Controls how many seconds to wait before shutting down. Sentry SDKs send events from a
+   * background queue and this queue is given a certain amount to drain pending events Default is
+   * 2000 = 2s
+   */
+  private long shutdownTimeout = 2000; // 2s
```

Comment:
I wouldn't rename any of the options not to bring any breaking changes here.

Also, I'd suggest we're always explicit on the unit. Since this is `long`, this timeout is in Seconds? MIlliseconds?

Joshua Bloch talks about that in this video which I highly recommend: https://www.youtube.com/watch?v=heh4OeB9A-c

Reply:
I agree. It got renamed because there was a mismatch between setters/getters and the field name, thus it didn't comply to Java Beans convention making it impossible to use with Spring `@ConfigurationProperties`. Either we rename the field (as it's done in this PR) and do not introduce breaking changes (as the setters & getters do not change), or follow Joshua Bloch suggestion and rename setters and getters.